.github/workflows/ci.yml

```diff
@@ -0,0 +1,187 @@
+  Javadoc:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'adopt'
+          cache: maven
+      - name: Generate javadocs
+        run: |
+          mvn clean install -DskipTests
+          mvn javadoc:javadoc -DskipTests
+  Run-E2E-Tests-1:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:latest
+        env:
+          POSTGRES_DB: fcfs
+          POSTGRES_PASSWORD: password
+          POSTGRES_USER: swirlds
+          PGDATA: /var/lib/postgresql/data/pgdata
+          POSTGRES_PORT: 5432
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    steps:
+      - uses: actions/checkout@v2 # checkout current branch
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'adopt'
+          cache: maven
+      - name: Build docker images # build the image for hedera-node
+        run: bash compose-build.sh
+      - name: Start network # start 3 node network
+        run: docker-compose -f docker-compose-ci.yml up -d
+      - name: Run E2E Suites # run selected e2e suites
+        run: |
+          mvn clean install -DskipTests
+          cd test-clients
+          mvn test -Dargs="-s contract.opcodes, contract.openzeppelin" -Pe2e-tests
+      - name: Download logfile
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: PackageSuiteRunnerReport
+          path: |
+            test-clients/src/main/java/com/hedera/services/bdd/suiterunner/logs/PackageSuitesRunner.log
+  Run-E2E-Tests-2:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:latest
+        env:
+          POSTGRES_DB: fcfs
+          POSTGRES_PASSWORD: password
+          POSTGRES_USER: swirlds
+          PGDATA: /var/lib/postgresql/data/pgdata
+          POSTGRES_PORT: 5432
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    steps:
+      - uses: actions/checkout@v2 # checkout current branch
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'adopt'
+          cache: maven
+      - name: Build docker images # build the image for hedera-node
+        run: bash compose-build.sh
+      - name: Start network # start 3 node network
+        run: docker-compose -f docker-compose-ci.yml up -d
+      - name: Run E2E Suites # run selected e2e suites
+        run: |
+          mvn clean install -DskipTests
+          cd test-clients
+          sleep 45s
```

Comment:
Sleeping is brittle. I assume this is to wait for docker containers to be up? We should add a health check to the docker container and use `docker compose -f docker-compose-ci.yml up --wait` to automatically wait for it to be ready before moving on.